Furnace count supplemental coil reporting

LandanT opened the pull request:
Pull request overview
---------------------

<!--- DESCRIBE PURPOSE OF THIS PULL REQUEST -->
This pull request is to update the comstock reporting measure for gas equipment in response to Issues #12. This measure now splits the supplemental and primary gas heating coils into two seperate counts. Updates were made to the measure and the comstock columns definitions .csv

 - Fixes #12

### Pull Request Author

This pull request makes changes to (select all the apply):
 - [ ] Documentation
 - [ ] Infrastructure (includes singularity image, buildstock batch, dependencies, continuous integration tests)
 - [ ] Sampling
 - [ ] Workflow Measures
 - [ ] Upgrade Measures
 - [x] Reporting Measures
 - [x] Postprocessing

Author pull request checklist:
<!--- Add to this list or remove from it as applicable.  This is a simple templated set of guidelines. -->
 - [x] Tagged the pull request with the appropriate label (documentation, infrastructure, sampling, workflow measure, upgrade measure, reporting measure, postprocessing) to help categorize changes in the release notes.
 - [x] Register values added to `comstock_column_definitions.csv`
 `test/resource_measure_tests.txt`
 - [ ] All new and existing tests pass the CI
 - [x] Updated measure .xml(s)
### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a code review on GitHub
 - [ ] All related changes have been implemented: data and method additions, changes, tests
 - [ ] If fixing a defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] Reviewed change documentation
 - [ ] Ensured code files contain License reference
 - [ ] Results differences are reasonable
 - [ ] Make sure the newly added measures has been added with tests and indexed properly
 - [ ] CI status: all tests pass

#### ComStock Licensing Language - Add to Beginning of Each Code File
```
# ComStock™, Copyright (c) 2023 Alliance for Sustainable Energy, LLC. All rights reserved.
# See top level LICENSE.txt file for license terms.
```
```

LandanT committed:
Updated Reporting Measure

Updated the reporting measure to split out primary and supplementary gas coil counts. This was done by testing if the model has a heating and cooling heat pump. If the model does it assumes that all NG Heating Coils in the model are supplemental heat.

LandanT committed:
Update measure.rb

Updated measure to split out secondary/supplemental gas heating coils from the primary heating counts. Supplementary gas coils are assumed to be all gas coils associated with HP equipment.

LandanT committed:
Update comstock_column_definitions.csv

Updated column definitions to reflect the changes in the reporting measures. Call all outputs (primary/supplemental) gas equipment or coils.

LandanT committed:
Update measure.xml

Updated Measure XML

mdahlhausen commented on measures/comstock_sensitivity_reports/measure.rb:
```diff
@@ -2719,55 +2719,107 @@ def run(runner, user_arguments)
-      elsif coil.autosizedNominalCapacity.is_initialized
-        capacity_w = coil.autosizedNominalCapacity.get
+      supplemental_capacity_w = 0.0
+      if !model.getAirLoopHVACUnitarySystems.empty? 
```
this selection criteria isn't going to work - 
1) some models may have secondary systems types that are AirLoopHVACUnitarySystems. If the primary system has gas coils, but has AirLoopHVACUnitarySystems, then all gas coils will be labeled as supplemental.
2) Some systems, particularly PSZ-AC RTUs, are modeled as AirLoopHVACUnitarySystems. They have primary gas coils, which will also be logged as supplementary here.

I suggest testing some models with these kinds of variations and seeing if the output variables are logging the capacities into the correct bins.

mdahlhausen replied:
perhaps the more direct method is to call .supplementalCoil and check if the coil name in the loop matches the coil on the systems (in essence, whether the coil is part of a unitary system, and is the supplementary coil on that system.

LandanT replied:
Matt, I have made some updates to the code to use the .supplementalCoil method. We can discuss in person 3/12

LandanT committed:
Update measure.rb

Fixed issues noted by matt and utilized .supplementalHeatingCoil to more directly test for supplemental coils.

LandanT committed:
Update measure.xml

LandanT committed:
Fixed issues identified

Updated measure to not use a do loop and instead directly test equipment.

mdahlhausen reviewed (approved):
This is working as intended, thanks!
We need to wait to merge until after the release. But once that's done, merge this first, then update the other PRs with it too. You'll need to re-generate the measure .xmls for those Prs.
